osquery/logger/plugins/http_logger.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant 
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <exception>
+#include <mutex>
+
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+#include <osquery/filesystem.h>
+#include <osquery/flags.h>
+#include <osquery/logger.h>
+#include <osquery/registry.h>
+#include <osquery/database.h>
+
+namespace osquery {
+
+FLAG(string,
+     logger_uri,
+     "Not set",
+     "Endpoint path for ERROR/WARN/INFO and results logging");
+
+//const std::string kFilesystemLoggerFilename = "osqueryd.results.log";
+
+std::mutex httpLoggerPluginMutex;
+
+class HTTPLoggerPlugin : public LoggerPlugin {
+ public:
+  Status setUp();
+  Status logString(const std::string& s);
+  Status init(const std::string& name, const std::vector<StatusLogLine>& log);
+  Status logStatus(const std::vector<StatusLogLine>& log);
+ private:
+  std::shared_ptr<DBHandle> db_;
```

Comment:
Although it may be a slight performance optimization to store the reference? I wonder.